### PR TITLE
Intro to git

### PR DESCRIPTION
**Pull request by frheault**
This is part 3 of the major refactor: Git

I decided to make the Intro to Git really about the concept and explaining why/how and defining terms.

Most of it came from the Git section in 'CODING STANDARDS IN THE LAB'.

TODO: Next week I will make sure the existing Git section in the 'CODING STANDARDS IN THE LAB' is an actual tutorial for Git and what we expect in the lab (how to handle forking, issues, PR, reviewing code, merging code, releases, following along tests in Github, fetching code to test locally during PR, etc)
This will act as guidelines for Scilpy specifically AND an advanced tutorial on good practices Git (mostly for scipy/dipy/nibabel repo in python)

### CODE REVIEW

**Comment by frheault** [comment]
@arnaudbore @AlexVCaron @EmmaRenauld What do you think of <intro to git> section?
(And my plan for the Git section specifically about Scilpy)

File: source/coding/git.rst [hunk 1]
```diff
@@ -1,153 +1,64 @@
...
+*If you are only a user and you don't plan on developing the code:*
 
-How we use git in the lab
--------------------------
+You can download a repository; we say to **clone** a repository. Downloading vs cloning: when you download, it takes the code as it is then, wraps it in a zip file and downloads it all. But then, you have nothing telling you where you downloaded it from and you can't update the code. Cloning keeps the link with git and allows you to update. You will need to open a terminal and use the :bash:`git clone` command. You can find the address by clicking on Github's green button *clone repository*.
 
-*If you are only a user and your don't plan on developping the code:*
+Once cloned we recommend either using a tag (a release) with the :bash:`git checkout` to ensure a stable version of the code. We expect users to remember which repository the cloned, to put the code in a appropriate folder on their computer (and know where it is), approximately when they cloned it and to remember that Scilpy constantly evolve.
```
Comment (EmmaRenauld) [inline source/coding/git.rst:16]: I'm not sure the first sentence is easy for someone not used to it. Maybe:

Once cloned we recommend you to ensure you work with a stable version of the code. Often, programmers will separate their most recent (less tested) code from more stable versions by adding tags (releases) to the various versions. To use a specific version / tag / release, use  :bash:`git checkout VERSION`.

File: source/coding/git.rst [hunk 2]
```diff
@@ -1,153 +1,64 @@
...
 
-    # From the local repo to your forked repo
-    git push origin myBranch  # Will automatically send it to the same branch name in your forked repo.
+By default all repositories have a main version of the code (a branch called *master* or *main*). When you are exploring new pieces of code, fixing bugs or tinkering in general you should always create a branch first (with a meaningful name) :bash:`git checkout -b my_branch`.
 
-    # From your forked repo to the upstream repo:
-    # Use the Pull Request button on internet
+If your modifications are useful, and you want to keep it safe, you should commit and push them. If your modifications could be useful to everyone and you want to share it with the rest of the world [2]_, you should consider doing a *pull request*. Only when everything is clean and tested, in the Github web interface, you can start an official pull request that will be reviewed by other members of the lab.
```
[the remark this answers is not in the document]
Comment (EmmaRenauld) [reply source/coding/git.rst:33]: If you want to [...], and you want the modifications on one branch of your own fork to be added to the real scilpy repo (upstream), you should consider ...

When that branch is accepted and merged to the master branch in the upstream, then if you update your own master branch (`git pull`), you will see the changes. You will then be able to delete safely your branch in your fork (`git branch -d my_test_branch`).

File: source/coding/git.rst [hunk 3]
```diff
@@ -1,153 +1,64 @@
...
 
-    # From your local repo to the upstream repo:
-    # DON'T DO THAT. Didn't you see that there is no such arrow on the figure!
+.. image:: /images/git_in_the_lab_features.png
+    :scale: 50 %
+    :align: left
 
+Generally, you should keep your master equal to the upstream's master. You should develop each feature in a separated branch. When you are ready to send your code to the upstream (pull request), it is usually not a good practice to send the work from your master branch, it should come from a branch with this purpose.
```
Comment (EmmaRenauld) [inline source/coding/git.rst:39]: Oh. So based on my comment above, I would bring this paragraph a bit higher.

**Review by EmmaRenauld — APPROVED** [review]
A few suggestions trying to think as someone who does not understand git 100% but to me it is clear.

**Pull request merged**